File: phm-server/src/main/java/com/phm/model/entity/PetSupply.java
package com.phm.model.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;

/**
 * 宠物用品实体类
 */
@Data
@TableName("pet_supply")
public class PetSupply {
    @TableId(type = IdType.AUTO)
    private Integer id;         // 主键ID
    private String name;        // 用品名称
    private String description; // 用品介绍
    private BigDecimal price;  // 用品价格
    private String shelfTime; // 上架时间
    private String imageFilename;    // 图片文件名
}

File: phm-ui/src/model/entiy/PetSupply.ts
// 宠物用品实体类型（对应Java的PetSupply）
import {PageQuery} from "@/model/VO/BackQuery.ts";

export interface PetSupply {
  id?: number;              // 主键ID（可选，新增时无ID）
  name?: string;             // 用品名称（必填）
  description?: string;     // 用品介绍（可选）
  price?: number;
  shelfTime?: string;       // 上架时间（格式：YYYY-MM-DD）
  imageFilename?: string;   // 图片文件名（可选）
}

// 宠物用品查询参数（继承PageQuery）
export interface PetSupplyQuery extends PageQuery {
  name?: string;             // 用品名称（模糊匹配）
  shelfTime?: string;       // 上架时间（精准匹配，格式：YYYY-MM-DD HH:mm:ss）
}
